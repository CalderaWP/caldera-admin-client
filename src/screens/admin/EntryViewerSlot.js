import React from 'react';
import FormAdminToolbar from './FormAdminToolbar';
import FormAdminMainView from './FormAdminMainView';
import AdminSlot from "./AdminSlot";
import types from "../../types";
import PropTypes from 'prop-types';
import {FormEntryViewer} from "../../components/EntryViewer/FormEntryViewer";
import FormAdminHelpView from './FormAdminHelpView';

export default class EntryViewerSlot extends AdminSlot {

	render() {
		const {form,entries} = this.props;

		return (
			<div>
				<FormAdminToolbar.NavBar
					label="Entry Viewer"
					onActive={this.handleActive}
					onDeactive={this.handleDeactive}
					isActive={this.state.active}
				/>
				{this.state.active &&
				<React.Fragment>

					<FormAdminMainView.Content>
						<FormEntryViewer
							form={form}
							entries={entries}
						/>
					</FormAdminMainView.Content>
					<FormAdminHelpView.Content>
						Help Content
					</FormAdminHelpView.Content>

				</React.Fragment>
				}
			</div>
		);
	}


}

/**
 * Prop tyeps for CreateFormSlot component
 * @type {{form: shim, entries: shim}}
 */
EntryViewerSlot.propTypes = {
	form: PropTypes.shape(types.formType),
	entries: PropTypes.shape(types.entriesType),
};

EntryViewerSlot.props = {
	form : {
		ID: '',
		name: ''
	},
	field_details: {
		order: {},
		entry_list: {}
	}
}
