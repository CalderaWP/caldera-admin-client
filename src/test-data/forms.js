export  const formwithIdCf1 = {
	ID: 'CF1',
	name: 'name One',
	field_details: {
		"order": {
			"firstName": {
				"id": "firstName",
				"label": "First Name"
			},
			"lastName": {
				"id": "lastName",
				"label": "Last Name"
			},
			"strongName": {
				"id": "strongName",
				"label": "Strong Bad"
			}
		},
		"entry_list": {
			"id": {
				"id": "id",
				"label": "ID"
			},
			"datestamp": {
				"id": "datestamp",
				"label": "Submitted"
			}
		}
	}

};
export const formWithIdCf2 =  {
	"ID": "cf2",
	"name": "Primary Contact Form",
	"fields": {
		"fld_8768091": {
			"ID": "fld_8768091",
			"name": "First Name",
			"type": "text"
		},
		"fld_9970286": {
			"ID": "fld_9970286",
			"name": "Last Name",
			"type": "text"
		},
		"fld_6009157": {
			"ID": "fld_6009157",
			"name": "Email Address",
			"type": "email"
		},
		"fld_7683514": {
			"ID": "fld_7683514",
			"name": "Comments Questions",
			"type": "paragraph"
		}
	},
	"emailIdentifyingFields": [],
	"piiFields": [],
	"privacyExporterEnabled": false,
	field_details: {
		"order": {
			"fld_8768091": {
				"id": "fld_8768091",
				"label": "First Name"
			},
			"fld_9970286": {
				"id": "fld_9970286",
				"label": "Last Name"
			},
			"fld_6009157": {
				"id": "fld_6009157",
				"label": "Email Address"
			},
			"fld_7683514": {
				"id": "fld_7683514",
				"label": "Comments Questions"
			}
		},
		entry_list: {
			"id": {
				"id": "id",
				"label": "ID"
			},
			"datestamp": {
				"id": "datestamp",
				"label": "Submitted"
			}
		}
	},
	"mailer": {
		"active": false
	},
	"entries": {
		count: 3
	}
};
/**
 * Mock form collection.
 */
export default {
	cf1: formwithIdCf1,
	cf2: formWithIdCf2
};