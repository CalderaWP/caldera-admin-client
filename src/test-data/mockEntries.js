export const mockEntries =  {
	"32": {
		"id": "32",
		"form_id": "CF5b412c2eafc90",
		"datestamp": "2018-07-07 21:12:25",
		"status": "active",
		"user": {
			"id": "1",
			"name": "admin",
			"email": "1@nom.com"
		},
		"fields": {
			"fld_8768091": {
				"id": "99",
				"entry_id": "32",
				"field_id": "fld_8768091",
				"slug": "first_name",
				"value": "Mike"
			},
			"fld_9970286": {
				"id": "100",
				"entry_id": "32",
				"field_id": "fld_9970286",
				"slug": "last_name",
				"value": "Corkum"
			},
			"fld_6009157": {
				"id": "101",
				"entry_id": "32",
				"field_id": "fld_6009157",
				"slug": "email_address",
				"value": "mike@hiroy.club"
			},
			"fld_7683514": {
				"id": "102",
				"entry_id": "32",
				"field_id": "fld_7683514",
				"slug": "comments_questions",
				"value": "Hi Mike\r\n\r\nLine 2\r\n\r\nLine 11"
			}
		},
		"meta": []
	},
	"31": {
		"id": "31",
		"form_id": "CF5b412c2eafc90",
		"datestamp": "2018-07-07 21:12:03",
		"status": "active",
		"user": {
			"id": "1",
			"name": "admin",
			"email": "1@nom.com"
		},
		"fields": {
			"fld_8768091": {
				"id": "94",
				"entry_id": "31",
				"field_id": "fld_8768091",
				"slug": "first_name",
				"value": "Roy"
			},
			"fld_9970286": {
				"id": "95",
				"entry_id": "31",
				"field_id": "fld_9970286",
				"slug": "last_name",
				"value": "Sivan"
			},
			"fld_6009157": {
				"id": "96",
				"entry_id": "31",
				"field_id": "fld_6009157",
				"slug": "email_address",
				"value": "roy@hiroy.club"
			},
			"fld_7683514": {
				"id": "97",
				"entry_id": "31",
				"field_id": "fld_7683514",
				"slug": "comments_questions",
				"value": "Hi Roy\r\n\r\nLine 2\r\n\r\nLine 3"
			}
		},
		"meta": []
	}
}