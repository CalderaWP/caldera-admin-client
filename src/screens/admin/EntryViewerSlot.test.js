import renderer from 'react-test-renderer';
import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import EntryViewerSlot from './EntryViewerSlot'
import forms,{formWithIdCf2} from '../../test-data/forms'
import {mockEntries} from "../../test-data/mockEntries";
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});
const handler = () => {
};


describe('Entry Slot component', () => {
	it('Renders empty div when not active and has no form', () => {
		expect(
			renderer.create(
				<EntryViewerSlot
					entries={{}}
				/>
			).toJSON()
		).toMatchSnapshot();
	});

	it('Renders empty div when not active and has form', () => {

		expect(
			renderer.create(
				<EntryViewerSlot
					form={formWithIdCf2}
				/>
			).toJSON()
		).toMatchSnapshot();
	});

	it('Renders empty div when not active and has form and entries', () => {

		expect(
			renderer.create(
				<EntryViewerSlot
					form={formWithIdCf2}
					entries={mockEntries}
				/>
			).toJSON()
		).toMatchSnapshot();
	});


	it('Renders when active and has no entries', () => {
		const component = shallow(
			<EntryViewerSlot
				form={formWithIdCf2}
				entries={{}}
			/>
		);

		component.setState({active: true});
		expect(
			toJson(component)
		).toMatchSnapshot();
	});
	it('Renders when active and has entries', () => {
		const component = shallow(
			<EntryViewerSlot
				form={formWithIdCf2}
				entries={mockEntries}
			/>
		);

		component.setState({active: true});
		expect(
			toJson(component)
		).toMatchSnapshot();
	});


});