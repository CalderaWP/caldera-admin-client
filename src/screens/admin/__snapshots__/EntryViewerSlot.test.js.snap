// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry Slot component Renders empty div when not active and has form 1`] = `<div />`;

exports[`Entry Slot component Renders empty div when not active and has form and entries 1`] = `<div />`;

exports[`Entry Slot component Renders empty div when not active and has no form 1`] = `<div />`;

exports[`Entry Slot component Renders when active and has entries 1`] = `
<div>
  <Component
    isActive={true}
    label="Entry Viewer"
    onActive={[Function]}
    onDeactive={[Function]}
  />
  <React.Fragment>
    <Component>
      <FormEntryViewer
        entries={
          Object {
            "31": Object {
              "datestamp": "2018-07-07 21:12:03",
              "fields": Object {
                "fld_6009157": Object {
                  "entry_id": "31",
                  "field_id": "fld_6009157",
                  "id": "96",
                  "slug": "email_address",
                  "value": "roy@hiroy.club",
                },
                "fld_7683514": Object {
                  "entry_id": "31",
                  "field_id": "fld_7683514",
                  "id": "97",
                  "slug": "comments_questions",
                  "value": "Hi Roy
          
          Line 2
          
          Line 3",
                },
                "fld_8768091": Object {
                  "entry_id": "31",
                  "field_id": "fld_8768091",
                  "id": "94",
                  "slug": "first_name",
                  "value": "Roy",
                },
                "fld_9970286": Object {
                  "entry_id": "31",
                  "field_id": "fld_9970286",
                  "id": "95",
                  "slug": "last_name",
                  "value": "Sivan",
                },
              },
              "form_id": "CF5b412c2eafc90",
              "id": "31",
              "meta": Array [],
              "status": "active",
              "user": Object {
                "email": "1@nom.com",
                "id": "1",
                "name": "admin",
              },
            },
            "32": Object {
              "datestamp": "2018-07-07 21:12:25",
              "fields": Object {
                "fld_6009157": Object {
                  "entry_id": "32",
                  "field_id": "fld_6009157",
                  "id": "101",
                  "slug": "email_address",
                  "value": "mike@hiroy.club",
                },
                "fld_7683514": Object {
                  "entry_id": "32",
                  "field_id": "fld_7683514",
                  "id": "102",
                  "slug": "comments_questions",
                  "value": "Hi Mike
          
          Line 2
          
          Line 11",
                },
                "fld_8768091": Object {
                  "entry_id": "32",
                  "field_id": "fld_8768091",
                  "id": "99",
                  "slug": "first_name",
                  "value": "Mike",
                },
                "fld_9970286": Object {
                  "entry_id": "32",
                  "field_id": "fld_9970286",
                  "id": "100",
                  "slug": "last_name",
                  "value": "Corkum",
                },
              },
              "form_id": "CF5b412c2eafc90",
              "id": "32",
              "meta": Array [],
              "status": "active",
              "user": Object {
                "email": "1@nom.com",
                "id": "1",
                "name": "admin",
              },
            },
          }
        }
        form={
          Object {
            "ID": "cf2",
            "emailIdentifyingFields": Array [],
            "entries": Object {
              "count": 3,
            },
            "field_details": Object {
              "entry_list": Object {
                "datestamp": Object {
                  "id": "datestamp",
                  "label": "Submitted",
                },
                "id": Object {
                  "id": "id",
                  "label": "ID",
                },
              },
              "order": Object {
                "fld_6009157": Object {
                  "id": "fld_6009157",
                  "label": "Email Address",
                },
                "fld_7683514": Object {
                  "id": "fld_7683514",
                  "label": "Comments Questions",
                },
                "fld_8768091": Object {
                  "id": "fld_8768091",
                  "label": "First Name",
                },
                "fld_9970286": Object {
                  "id": "fld_9970286",
                  "label": "Last Name",
                },
              },
            },
            "fields": Object {
              "fld_6009157": Object {
                "ID": "fld_6009157",
                "name": "Email Address",
                "type": "email",
              },
              "fld_7683514": Object {
                "ID": "fld_7683514",
                "name": "Comments Questions",
                "type": "paragraph",
              },
              "fld_8768091": Object {
                "ID": "fld_8768091",
                "name": "First Name",
                "type": "text",
              },
              "fld_9970286": Object {
                "ID": "fld_9970286",
                "name": "Last Name",
                "type": "text",
              },
            },
            "mailer": Object {
              "active": false,
            },
            "name": "Primary Contact Form",
            "piiFields": Array [],
            "privacyExporterEnabled": false,
          }
        }
      />
    </Component>
    <Component>
      Help Content
    </Component>
  </React.Fragment>
</div>
`;

exports[`Entry Slot component Renders when active and has no entries 1`] = `
<div>
  <Component
    isActive={true}
    label="Entry Viewer"
    onActive={[Function]}
    onDeactive={[Function]}
  />
  <React.Fragment>
    <Component>
      <FormEntryViewer
        entries={Object {}}
        form={
          Object {
            "ID": "cf2",
            "emailIdentifyingFields": Array [],
            "entries": Object {
              "count": 3,
            },
            "field_details": Object {
              "entry_list": Object {
                "datestamp": Object {
                  "id": "datestamp",
                  "label": "Submitted",
                },
                "id": Object {
                  "id": "id",
                  "label": "ID",
                },
              },
              "order": Object {
                "fld_6009157": Object {
                  "id": "fld_6009157",
                  "label": "Email Address",
                },
                "fld_7683514": Object {
                  "id": "fld_7683514",
                  "label": "Comments Questions",
                },
                "fld_8768091": Object {
                  "id": "fld_8768091",
                  "label": "First Name",
                },
                "fld_9970286": Object {
                  "id": "fld_9970286",
                  "label": "Last Name",
                },
              },
            },
            "fields": Object {
              "fld_6009157": Object {
                "ID": "fld_6009157",
                "name": "Email Address",
                "type": "email",
              },
              "fld_7683514": Object {
                "ID": "fld_7683514",
                "name": "Comments Questions",
                "type": "paragraph",
              },
              "fld_8768091": Object {
                "ID": "fld_8768091",
                "name": "First Name",
                "type": "text",
              },
              "fld_9970286": Object {
                "ID": "fld_9970286",
                "name": "Last Name",
                "type": "text",
              },
            },
            "mailer": Object {
              "active": false,
            },
            "name": "Primary Contact Form",
            "piiFields": Array [],
            "privacyExporterEnabled": false,
          }
        }
      />
    </Component>
    <Component>
      Help Content
    </Component>
  </React.Fragment>
</div>
`;
